### DISCUSSION TITLE
Teacher refuses to migrate from fastpages because quarto lacks features relating to html and vanilla javascript

### DISCUSSION THREAD

**Discussion by moonpiedumplings** [quarto-dev/quarto-cli#4016] [Q&A]
Quarto is cool. I created a blog with it. I can run jupyter notebooks. However, I can't seem to figure out how to embed html into the markdown files. There is observable js, but it isn't compatible with vanilla javascript, and it is very new, which is part of why my teacher doesn't want to learn it, or force people to learn it. However, fastpages is falling apart, and is not maintainable by one classroom of people. We are currently using a jekyll static site as a frontend, but I would rather use quarto, because quarto is much easier to maintain and deploy than jekyll (really new m1 macs, and Ubuntu 22 on wsl are broken without using containerization or nix to deploy the server). 

https://nighthawkcoders.github.io/APCSP/frontend/binary

https://github.com/nighthawkcoders/APCSP/blob/f3e9bc862f20702d7965e8be3013eb7f738b08da/_posts/2022-07-07-PBL-binary.md?plain=1

https://nighthawkcoders.github.io/APCSP/data/jokes

https://github.com/nighthawkcoders/APCSP/blob/2688431ada980eec62b6d90424de8ac555593f26/_posts/2022-07-10-PBL-jokes.md?plain=1

Here are some example of what my teacher uses fastpages for. I know that it is theoretically possible to replicate all of this using observable js, but is there a way to just keep what code we have, without having to rewrite code? Is there a way to just embed html into the markdown file like in the above examples? Is there any way to use vanilla js and html?

I tried searching for similar discussions, but I couldn't find the right keywords or something, because everything I found was relating to the generation/output of html files or observable js.

**Comment by cscheid** [comment]
You can use raw HTML blocks:

````
```{=html}
<!-- the content of this block will be included as is in html formats, and nowhere else -->
<blink>You can even do this, though you shouldn't</blink>
<script>alert("and this, too. though you shouldn't, either")</script>
```
````

You can add [header includes as well](https://quarto.org/docs/output-formats/html-basics.html#includes)

**Reply by moonpiedumplings** [reply]
Yeah, I discovered this after I made this post. However, my teacher is still unsatisfied even after I presented this to him, and he would still rather use the deprecated fastpages. 

In his own words: 

"There are other things, like the ability to interact with liquid syntax and passing jekyll variables to make tagging and schedule.  Fundamentally, this thing has turned into a blogging thing vs a GitHub Pages extension.    Fastpages won’t break, as I have pulled it, as it really is just capitalizing on features of GitHub Pages.  I still want to make my own, that does not loose conformance to GitHub pages themes.  This way it is just a GitHub pages development platform.  Exactly what I want!  Very industry standard!."

He views fastpages as an extension to github pages that brings extra features, allowing github pages to act as a proper frontend, but I think those features can be replicated in quarto.

Although tags and categories can be replicated in quarto, I don't know if jekyll variables can.

https://nighthawkcoders.github.io/APCSP/schedule

https://github.com/nighthawkcoders/APCSP/blob/6ee1edc0d173deb5b172a2cd88caef2930892953/_pages/00_schedule.md?plain=1

Is there any way to replicate the above with quarto? I played around with the development javascript for liquid:

```
<!--for development-->
<script src="https://cdn.jsdelivr.net/npm/liquidjs/dist/liquid.browser.umd.js"></script>

```

But I couldn't get anything working. I'm just a student, and I haven't actually really worked with javascript, mainly other stuff.

**Reply by cscheid** [reply]
> ... just capitalizing on features of GitHub Pages. I still want to make my own, that does not loose conformance to GitHub pages themes

jekyll will be pretty much the only solution that satisfies this constraint, so I think quarto is not the right tool in this context.
